src/sp/it/pl/util/system/Environment.kt

```diff
@@ -82,71 +81,62 @@ fun File.runAsProgram(vararg arguments: String): Fut<Try<Void, Exception>> = run
  * @param command see [Runtime.exec]
  */
 @JvmOverloads
-fun runCommand(command: String, then: Consumer<Process>? = null) {
+fun runCommand(command: String, then: ((Process) -> Unit)? = null) {
     runNew {
         try {
             val p = Runtime.getRuntime().exec(command)
-            then?.accept(p)
+            then?.invoke(p)
         } catch (e: IOException) {
             logger.error(e) { "Error running command '$command'" }
         }
     }
 }
 
 /**
- * Browse the file in OS' file browser:
- * * if denotes a directory it will be opened
- * * if denotes a file, it's location will be opened and the file selected
- *
- * On some platforms the operation may be unsupported. In that case this method is a no-op.
+ * Browse the file in the systems file browser by opening the parent directory and selecting it.
+ * If it is unsupported, it will try to [open] the parent directory instead.
  */
-fun File.browse() = toURI().browse()
+fun File.browse() {
```

Comment:
This is wrong, because you just turned the method into a blocking call. Im removing the method, because if it would take the same pattern of running new thread, then ```URI.browse()``` would create 2 threads for URI that denotes a file.

Reply:
I mean im reverting back to approach of toURI.browse()

Reply:
The hint already suggesting the incorrectness is the parallel with OOP. File.browse() should rely on implementation of URI.browse(), not vice versa, because File is denoted by URI. FIle is URI. URI is not File. Browse is a function of URI, not File.